0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+| `additional_info`      | Object               | _(Optional)_ Arbitrary fields attached to this transfer. (For example, the IDs of related transfers in other systems.) |
+| `memo`                 | Object               | _(Optional)_ Arbitrary data provided by the `debit_account` for the `credit_account`. The connector stores the ILP Header in this field. |
+| `note_to_self`         | Object               | _(Optional)_ Arbitrary data provided by the `debit_account` for itself. This field MUST be hidden when not authenticated as the `debit_account` or an admin. |
+| `id`                   | [URL][]              | _(Server-provided)_ Primary resource identifier for this transfer. MUST be an HTTP(S) URL where you can [get the transfer resource][Get Transfer]. |
+| `fulfillment`          | [URL][]              | (Server-provided) Path to the fulfillment for this transfer. MUST be an HTTP(S) URL where the client can [submit the fulfillment][Submit Fulfillment] or [get the fulfillment][Get Transfer Fulfillment]. MUST be provided if and only if this transfer has an `execution_condition`. |
+| `transfer_rejection`   | [URL][]              | (Server-provided) Path where a client can [reject this transfer][Reject Transfer]. MUST be an HTTP(S) URL. |
+| `rejection_reason`     | String               | _(Server-provided)_ The reason the transfer was rejected. MUST appear if and only if `state` is `rejected`. |
+| `state`                | String               | _(Server-provided)_ The current state of the transfer. Valid states are `prepared`, `executed`, and `rejected`. |
+| `timeline`             | Object               | _(Server-provided)_ Timeline of the transfer's state transitions. |
+| `timeline.executed_at` | [Date-Time][]        | _(Server-provided)_ Time when the transfer was originally executed. MUST appear if and only if `state` is `executed`. This time MUST be equal to or later than the `prepared_at` time. |
+| `timeline.prepared_at` | [Date-Time][]        | _(Server-provided)_ Time when the transfer was originally prepared. MUST appear, even if the transaction is unconditional. |
+| `timeline.rejected_at` | [Date-Time][]        | _(Server-provided)_ Time when the transfer was originally rejected. MUST appear if and only if `state` is `rejected`. This time MUST be equal to or later than the `prepared_at` time. |
+
+[UUID]: http://en.wikipedia.org/wiki/Universally_unique_identifier
+
+
+#### Client IDs
+
+The transfer's `client_id` field is one of the main ways of identifying the transfer; it is unique across the ledger, so implementations MAY use it as a primary unique key in a database. The client applications, not the server, generate the `client_id`, so the server SHOULD require that the field match the canonical form for a [UUID][], which is 32 lowercase hexadecimal digits, separated by hyphens into groups of 8-4-4-4-12.
```

Comment:
> The reason to use an id like this within one ledger is to prevent people from accidentally submitting the same transfer twice. If the transfer with that ID has already been submitted, POSTing it again will cause an error.

> The client_id should not be kept the same across ledgers because of the potential for id squatting. If you are a connector that always uses the same id for the outgoing transfer as the incoming one, I can prevent you from executing the destination transfer if I know the source transfer's id (for example if I am the sender).

There are some normative requirements there that we should document in the spec.